frontend/src/components/explorer/mapViewer/SidebarDataPanels.vue

```diff
@@ -160,13 +218,62 @@ export default {
       showMapCardContent: true,
       showSelectionCardContent: true,
       messages,
+      showModal: false,
+      showFullReactionListMissing: false,
+      showFullReactionListMap: false,
+      displayedReaction: 40,
+      limitReaction: 999999999,
+      missingNumberOfReactions: null,
     };
   },
   computed: {
     ...mapState({
       model: state => state.models.model,
       loading: state => state.maps.loadingElement,
+      mapReactionList: state => state.maps.svgReactionsIdList,
     }),
+    modelNumberOfReactions() {
+      return this.currentMap.reactionList ? this.currentMap.reactionList.length : null;
+    },
+    mapReactionIdListHtml() {
+      const l = ['<span class="tags">'];
+      for (let i = 0; i < this.mapReactionList.length; i += 1) {
+        const r = this.mapReactionList[i];
+        if ((!this.showFullReactionListMap && i === this.displayedReaction)
+          || i === this.limitReaction) {
+          break;
+        }
+        const customLink = buildCustomLink({ model: this.model.short_name, type: 'reaction', id: r, title: r, cssClass: 'target="_blank"' });
+        l.push(
+          `<span id="${r}" class="tag">${customLink}</span>`
+        );
+      }
+      l.push('</span>');
+      return l.join('');
```

Comment:
Could this be don in a Vue template/component instead?

Reply:
This pattern is also prevalent in other places such as [Subsystem.vue](https://github.com/MetabolicAtlas/MetabolicAtlas/blob/4e46aeea9605df5040c7bc02aa9c2d97bfebb878/frontend/src/components/explorer/gemBrowser/Subsystem.vue#L139). I remember there was a problem with performance when trying to render too many `router-link` elements but not sure if the problem was exactly here. Either way we should probably open up a new issue if we want to consider having these in template/component.

Reply:
>open up a new issue 

@e0 or @nanjiangshu please do so, and feel free to link to the above comment of this PR. I'm marking this discussion as resolved.